dialogue-core/src/main/java/com/palantir/dialogue/core/NodeSelectionStrategyChannel.java

```diff
@@ -93,8 +95,16 @@ static LimitedChannel create(Config cf, ImmutableList<LimitedChannel> channels)
 
     @Override
     public Optional<ListenableFuture<Response>> maybeExecute(Endpoint endpoint, Request request) {
-        Optional<ListenableFuture<Response>> maybe = delegate.maybeExecute(endpoint, request);
-        if (!maybe.isPresent()) {
+        HostId hostId = request.attachments().getOrDefault(RoutingAttachments.EXECUTE_ON_HOST_ID_KEY, null);
```

Comment:
I'm afraid this might skip some metrics gathering and updating, so likely all the strategies need to be HostId aware and do sensible things